benchmarking/compute_probability_of_improvement.py

```diff
@@ -0,0 +1,251 @@
+        A tuple of:
+         - probability of improvement
+         - confidence interval
+         - number of samples per env
+         - number of baseline samples per env
+    """
+    envs = runs_by_env.keys()
+    baseline_envs = baseline_runs_by_env.keys()
+    comparison_envs = sorted(set(envs).intersection(set(baseline_envs)))
+
+    run_scores = sample_matrix_from_runs_by_env(runs_by_env, comparison_envs)
+    baseline_run_scores = sample_matrix_from_runs_by_env(
+        baseline_runs_by_env,
+        comparison_envs,
+    )
+    samples_per_env = run_scores.shape[0]
+    baseline_samples_per_env = baseline_run_scores.shape[0]
+
+    probabs, error_intervals = rly.get_interval_estimates(
+        {"baseline_vs_new": (baseline_run_scores, run_scores)},
+        metrics.probability_of_improvement,
+        reps=reps,
+    )
+    probability_of_improvement = probabs["baseline_vs_new"]
+    confidence_interval = np.squeeze(error_intervals["baseline_vs_new"])
+
+    return (
+        probability_of_improvement,
+        confidence_interval,
+        samples_per_env,
+        baseline_samples_per_env,
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("runs_dir", type=pathlib.Path)
+    parser.add_argument("baseline_runs_dir", nargs="?", default=None, type=pathlib.Path)
+    parser.add_argument("--baseline-algo", type=str)
+    parser.add_argument("--algo", type=str)
+    parser.add_argument("--bootstrap-reps", type=int, default=2000)
+
+    args = parser.parse_args()
+
+    if args.baseline_runs_dir is None:
+        args.baseline_runs_dir = args.runs_dir
+
+    runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.runs_dir,
+        only_completed_runs=True,
+    )
+    baseline_runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.baseline_runs_dir,
+        only_completed_runs=True,
+    )
+
+    algos = sorted(runs_by_algo_and_env.keys())
+    baseline_algos = sorted(baseline_runs_by_algo_and_env.keys())
+
+    try:
+        if len(algos) == 0:
+            raise ValueError(f"The run directory [{args.runs_dir}] contains no runs.")
+
+        if len(baseline_algos) == 0:
+            raise ValueError(
+                f"The baseline run directory [{args.baseline_runs_dir}] "
+                f"contains no runs.",
+            )
+
+        if "algo" not in args is None:
+            if len(algos) == 1:
+                args.algo = algos[0]
+            else:
+                raise ValueError(
+                    f"The run directory [{args.runs_dir}] contains runs for the "
+                    f"algorithms [{', '.join(algos)}]. Please use the --algo option "
+                    f" to specify which algorithms runs to compare.",
+                )
+
+        if args.baseline_algo is None:
+            if len(baseline_algos) == 1:
+                args.baseline_algo = baseline_algos[0]
+            elif args.algo in baseline_algos:
+                args.baseline_algo = args.algo
```

Comment:
What's the intent here?

Reply:
(it seems like we're comparing an algorithm to itself in this branch; is this ever a useful operation?)